YapDatabase/YapDatabase.m

```diff
@@ -158,7 +158,10 @@ + (YapDatabaseSerializer)defaultSerializer
 + (YapDatabaseDeserializer)defaultDeserializer
 {
 	return ^ id (NSString __unused *collection, NSString __unused *key, NSData *data){
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 		return data && data.length > 0 ? [NSKeyedUnarchiver unarchiveObjectWithData:data] : nil;
+#pragma clang diagnostic pop
```

Comment:
Is this to get around the fact that we do not know the type of the class at deserialisation time?

Reply:
Annoyingly `unarchiveObjectWithData:` doesn't exist beyond iOS 12, so we'll need to actually change this code soon, if we cannot drop YapDatabase :(

Reply:
That's right.
Probably not super hard to add the types if we really do want to keep YapDB. But I didn't want to waste time doing it here.